src/views/build.ts

```diff
@@ -0,0 +1,125 @@
+        }
+
+        const ans = await vscode.window.showErrorMessage("Build failed, do you want to continue?",
+            "Proceed", "Fix...", "Cancel");
+        sendInfo(operationId, {
+            operationName: "build",
+            choiceForBuildError: ans || "esc",
+        });
+        if (ans === "Proceed") {
+            return true;
+        } else if (ans === "Fix...") {
+            showFixSuggestions(operationId);
+        }
+        return false;
+    }
+    return false;
+}
+
+function checkErrorsReportedByJavaExtension(): boolean {
+    const problems = vscode.languages.getDiagnostics() || [];
+    for (const problem of problems) {
+        const fileName = path.basename(problem[0].fsPath || "");
+        if (fileName.endsWith(".java") || fileName === "pom.xml" || fileName.endsWith(".gradle")) {
+            if (problem[1].filter((diagnostic) => diagnostic.severity === vscode.DiagnosticSeverity.Error).length) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+export const BUILD_FAILED = "build-failed-do-you-want-to-continue";
+
+async function showFixSuggestions(operationId: string) {
+    let buildFiles = [];
+    try {
+        buildFiles = await lsPlugin.resolveBuildFiles();
+    } catch (error) {
+        // do nothing
+    }
+
+    const pickitems = [];
+    pickitems.push({
+        label: "Clean workspace cache",
+        detail: "Clean the stale workspace and reload the window",
+    });
+    if (buildFiles.length) {
+        pickitems.push({
+            label: "Update project configuration",
+            detail: "Force the language server to update the project configuration/classpath",
+        });
+    }
+    pickitems.push({
+        label: "Open log file",
+        detail: "Open log file to view more details for the build errors",
+    });
+    pickitems.push({
+        label: "Troubleshooting guide",
+        detail: "Find more detail about the troubleshooting steps",
+    });
+
+    const ans = await vscode.window.showQuickPick(pickitems, {
+        placeHolder: "Please fix the errors in PROBLEMS first, then try the fix suggestions below.",
+    });
+    sendInfo(operationId, {
+        operationName: "build",
+        choiceForBuildFix: ans ? ans.label : "esc",
+    });
+    if (!ans) {
+        return;
+    }
+
+    if (ans.label === "Clean workspace cache") {
+        vscode.commands.executeCommand("java.clean.workspace");
+    } else if (ans.label === "Update project configuration") {
+        for (const buildFile of buildFiles) {
+            await vscode.commands.executeCommand("java.projectConfiguration.update", vscode.Uri.parse(buildFile));
+        }
+    } else if (ans.label === "Open log file") {
+        vscode.commands.executeCommand("java.open.serverLog");
+    } else if (ans.label === "Troubleshooting guide") {
+        utility.openTroubleshootingPage("Build failed", BUILD_FAILED);
```

Comment:
i see you just jump to the debugger troubleshooting page, a little weird from a user perspective. If you don't define your own troubleshooting in the dependency view, you can remove this option from the Fix quickpicks.